contributors/design-proposals/componentconfig-dump.md

```diff
@@ -0,0 +1,164 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should refer to the docs that go with that version.
+
+Documentation for other releases can be found at [releases.K8s.io](http://releases.K8s.io).
+</strong>
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+
+# Overview
+
+We describe a proposal for retrieving component related configurations (componentconfigs and otherwise) that govern the behavior of any one of the Kubernetes (K8s) components.
+
+We propose several important changes:
+
+* adding new authenticated protected path to a component api:
+    * `/configs`
```

Comment:
I think you want to know two things:
- intended config (e.g. an object you put on the API server)
- actual config (what the component is using *right now*)

Having a way of exposing both of these is part of the to dynamic config discussion: https://github.com/kubernetes/kubernetes/pull/29459

The latter case is what an endpoint on a given component could be used for.
Another option is to expose this information via a monitoring system like Prometheus, and achieve security by controlling access to your monitoring system.